sklearn/metrics/tests/test_classification.py

```diff
@@ -1579,3 +1581,40 @@ def test_brier_score_loss():
     # calculate even if only single class in y_true (#6980)
     assert_almost_equal(brier_score_loss([0], [0.5]), 0.25)
     assert_almost_equal(brier_score_loss([1], [0.5]), 0.25)
+
+
+def test_check_labels_subset():
+    y = [1, 1, 2, 2, 3, 3]
+    labels1 = np.array([1, 2])
+    labels2 = np.array([1, 2, 4])
+    expected_msg1 = ("The `allow_labels_subset` argument should be True "
+                     "if a subset of available classes=[1 2 3] was passed in "
+                     " the labels=[1 2] argument. `allow_labels_subset` was "
+                     " added in version 0.20 and this behavior will "
+                     "result in error from version 0.22.")
+    expected_msg2 = ("The `allow_labels_subset` argument should be True "
+                     "if a subset of available classes=[1 2 3] was passed in "
+                     " the labels=[1 2 4] argument. `allow_labels_subset` was "
+                     " added in version 0.20 and this behavior will "
+                     "result in error from version 0.22.")
+
+    scorers_to_check = [confusion_matrix, cohen_kappa_score, recall_score,
+                        precision_score, precision_recall_fscore_support,
+                        fbeta_score, f1_score, classification_report,
+                        hamming_loss, log_loss]
+
+    for scorer in scorers_to_check:
+        kwargs = {}
+        params = signature(scorer).parameters
+        if 'average' in params:
+            kwargs['average'] = 'micro'
+        if 'beta' in params:
+            kwargs['beta'] = 2
+
+        assert_warns_message(DeprecationWarning,
+                             expected_msg1,
+                             scorer, y, y, labels=labels1, **kwargs)
+        if scorer != log_loss:
```

Comment:
Add test that with ``allow_label_subset`` there is no warning.

Reply:
> why?

because log_loss will not work if the labels argument has more classes than what's present in the data. so case 2 cannot be tested.

> Add test that with allow_label_subset there is no warning.

will do